executor/builder.go

```diff
@@ -970,7 +969,7 @@ func (b *executorBuilder) buildExplain(v *plannercore.Explain) Executor {
 	}
 	if v.Analyze {
 		if b.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl == nil {
-			b.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl = execdetails.NewRuntimeStatsColl()
+			b.ctx.GetSessionVars().StmtCtx.InitRuntimeStatsColl()
 		}
```

Comment:
Maybe changes in tests come from here, `RuntimeStatsColl` won't be nil after this PR. And it is not re-inited if `!globalConfig.EnableCollectExecutionInfo == true`. @tiancaiamao

Reply:
Now I don't reset `RuntimeStatsColl` for the `ExplainFor` cases